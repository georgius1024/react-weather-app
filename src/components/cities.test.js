import React from 'react'
import jest from 'jest-mock'
import renderer from 'react-test-renderer'
import Cities from './cities'
const cities = [
  {
    country: 'RU',
    name: 'Moscow',
    lat: '55.75222',
    lng: '37.61556',
    image: '/images/11.jpg',
    temperature: '-7',
    humidity: '78',
    rain: '0',
    snow: '10',
    cloudiness: '80',
  },
]
describe('Cities', () => {
  it('renders correctly', () => {
    const onAdd = jest.fn()
    const selected = cities[0]
    const DOM = renderer.create(
      <Cities cities={cities} selected={selected} onAdd={onAdd} />
    )
    const html = JSON.stringify(DOM, null, 2)
    expect(html).toContain(selected.name)
    expect(DOM).toMatchSnapshot()
  })
})
