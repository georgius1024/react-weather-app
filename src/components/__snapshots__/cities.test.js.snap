// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities renders correctly 1`] = `
<aside
  className="menu"
>
  <p
    className="menu-label"
  >
    Favorite cities
  </p>
  <ul
    className="menu-list"
  >
    <li>
      <a
        className="is-active"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Moscow
      </a>
    </li>
  </ul>
  <div
    className="menu-label"
  >
    <hr />
  </div>
  <ul
    className="menu-list"
  >
    <li>
      <a
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <i
            className="fas fa-plus"
          />
        </span>
        Add city
      </a>
    </li>
  </ul>
</aside>
`;
